packages/funman/core/taskrunner.py

```diff
@@ -0,0 +1,133 @@
+# FIXME: this is a copy from mira, should consolidate
+import json
+import signal
+import sys
+import concurrent.futures
+import argparse
+import os
+import time
+import threading
+from typing import Callable
+
+
+READ_CHUNK_SIZE = 1024*1024
+
+
+class SelfDestructThread:
+    def __init__(self, self_destruct_timeout_seconds):
+        self.self_destruct_timeout_seconds = self_destruct_timeout_seconds
+        self.stop_destruct_event = threading.Event()
+        self.thread = threading.Thread(target=self._kill_after_timeout)
+
+    def start(self):
+        self.thread.start()
+
+    def stop(self):
+        self.stop_destruct_event.set()
+        self.thread.join()
+
+    def _kill_after_timeout(self):
+        self.stop_destruct_event.wait(self.self_destruct_timeout_seconds)
+        if not self.stop_destruct_event.is_set():
+            # if for whatever reason this process is still around, do our best to self destruct
+            os.kill(
+                os.getpid(), signal.SIGKILL
+            )  # Send the SIGTERM signal to the current process
+
+
+class TaskRunnerInterface:
```

Comment:
@kbirk this is a carbon-copy of the taskrunner.py from mira-taskrunner, any thoughts on if we want to consolidate?

Reply:
Something like taskrunner core for python, for julia, for c++ etc that can be importer instead of defined on each packages folder?

Reply:
Just logging this into a ticket here:

https://github.com/DARPA-ASKEM/terarium/issues/3395